superset-frontend/cypress-base/cypress/integration/dashboard/drilltodetail.test.ts

```diff
@@ -0,0 +1,294 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import {
+  waitForChartLoad,
+  ECHARTS_CHARTS,
+  ECHARTS_DASHBOARD,
+} from './dashboard.helper';
+
+function interceptSamples() {
+  cy.intercept(`/datasource/samples*`).as('samples');
+}
+
+describe('Drill to detail modal', () => {
+  beforeEach(() => {
+    cy.login();
+    cy.visit(ECHARTS_DASHBOARD);
+    ECHARTS_CHARTS.forEach(waitForChartLoad);
+  });
+
+  it('opens the modal from the context menu', () => {
+    cy.get(
```

Comment:
Can we extract the code that opens the modal to a function? Specifically this part:

```
cy.get(
  "[data-test-viz-type='big_number_total'] [aria-label='More Options']",
).click();
cy.get('.ant-dropdown')
  .not('.ant-dropdown-hidden')
  .find("[role='menu'] [role='menuitem']")
  .eq(5)
  .should('contain', 'Drill to detail')
  .click();
```

Reply:
+1

Reply:
Sure. Updated